general/restrict-hostnames/rego/src_test.rego

```diff
@@ -1242,8 +1291,132 @@ test_allowed_vs_hostname_conflicts {
+		"review": {
+			"kind": {
+				"group": "networking.k8s.io",
+				"kind": "Ingress",
+			},
+			"operation": "CREATE",
+			"userInfo": {
+				"groups": null,
+				"username": "alice",
+			},
+			"object": {
+				"metadata": {
+					"name": "unallowed-ingress",
+					"namespace": "test",
+				},
+				"spec": {"rules": [{
+					"host": "unallowedtest.com",
+					"http": {"paths": [{
+						"pathType": "ImplementationSpecific",
+						"backend": {"service": {
+							"name": "unallowedtest",
+							"port": {"number": 443},
+						}},
+					}]},
+				}]},
+			},
+		},
+	}
+
+	result := violation with input as ingress with input.parameters.errorMsgAdditionalDetails as "(Additional details placeholder)"
+
+	# Violation expected
+	print(result)
+	count(result) > 0
+}
+
+# Test for VirtualService with unallowed host and no path
+test_vs_unallowed_host_no_path {
+	vs := {
+		"apiVersion": "admission.k8s.io/v1beta1",
+		"kind": "AdmissionReview",
+		"review": {
+			"kind": {
+				"group": "networking.istio.io",
+				"kind": "VirtualService",
+			},
+			"operation": "CREATE",
+			"userInfo": {
+				"groups": null,
+				"username": "alice",
+			},
+			"object": {
+				"metadata": {
+					"name": "unallowed-vs",
+					"namespace": "test",
+				},
+				"spec": {
+					"hosts": ["nope.com", "nah.com", "ok.test.com"],
+					"http": [{"route": [{"destination": {"host": "myservice"}}]}],
+				},
+			},
+		},
+	}
+
+	exemptions := ["*.test.com"]
+
+	result := violation with input as vs with input.parameters.exemptions as exemptions with input.parameters.errorMsgAdditionalDetails as "(Additional details placeholder)"
+
+	# Violation expected
+	print(result)
+	count(result) > 0
+}
+
+# Test for an allowed VS which sets a host but not matches on the path.
+# This is effectively the same as `test.com` is effectively the same as `test.com/` since the `/` is a seperator in our prefix strategy.
+test_vs_allowed_no_path {
+	vs_review := {
+		"apiVersion": "admission.k8s.io/v1beta1",
+		"kind": "AdmissionReview",
+		"review": {
+			"kind": {
+				"group": "networking.istio.io",
+				"kind": "VirtualService",
+			},
+			"operation": "CREATE",
+			"userInfo": {
+				"groups": null,
+				"username": "alice",
+			},
+			"object": {
+				"metadata": {
+					"name": "test",
+					"namespace": "test",
+				},
+				"spec": {
+					"hosts": ["test.com"],
+					"http": [],
+				},
+			},
+		},
+	}
+
+	namespaces := {"test": {
+		"apiVersion": "v1",
+		"kind": "Namespace",
+		"metadata": {
+			"annotations": {"ingress.statcan.gc.ca/allowed-hosts": `[{"host": "test.com","path":"/"}]`},
+			"name": "test",
+		},
+	}}
+
+	exemptions := [""]
+
+	result := violation with input as vs_review with data.inventory.cluster.v1.Namespace as namespaces with input.parameters.exemptions as exemptions with input.parameters.errorMsgAdditionalDetails as "(Additional details placeholder)"
 
-	#Empty set means no violations
+	# Empty set means no violations
 	result == set()
 }
```

Comment:
I've added this test case which should pass since it is `test.com` and `test.com/` is effectively the same for our rules. Dunno if we want to address this or just make a note and fix it in another ticket.

Reply:
@frazs @MichaelPatsula

Reply:
Per the latest commit, modifying the new violations (and existing Ingress violation modified with `object.get`) so that when no path is found, the path argument is passed as `"/"` instead of `""` appears to satisfy this test case without messing up any of the other ones

Reply:
Also confirmed that even with the change to the argument, this new test case is still happy if the allowed annotation uses `""` instead of `"/"` (which I think is a case `test_ingress_empty_path` already handles, so it probably doesn't warrant a new test case).

Reply:
For the no path edge case, currently there are the test cases:
- test_ingress_unallowed_host_no_path
- test_vs_unallowed_host_no_path 
- test_vs_allowed_no_path

Maybe we should also have a test case for test_ingress_allowed_no_path.

Reply:
In with the latest commit, if it looks good this can be resolved and the PR merged 😄